rollout/canary.go

```diff
@@ -248,21 +242,81 @@ func (c *rolloutContext) scaleDownOldReplicaSetsForCanary(oldRSs []*appsv1.Repli
 	return totalScaledDown, nil
 }
 
-func (c *rolloutContext) replicaSetReferencedByCanaryTraffic(rs *appsv1.ReplicaSet) bool {
+// isReplicaSetReferenced returns if the given ReplicaSet is still being referenced by any of
+// the current, stable, blue-green services. Used to determine if the ReplicaSet can
+// safely be scaled to zero, or deleted.
+func (c *rolloutContext) isReplicaSetReferenced(rs *appsv1.ReplicaSet) bool {
```

Comment:
Would it make sense to move this to the rollout/context.go since it handles both bluegreen and canary?

Reply:
Yes, that would make more sense. Will adjust.

Reply:
I ended up moving it to replicaset.go since it's called `isReplicaSetReferenced`